cs3api4lab/tests/test_cs3apismanager.py

```diff
@@ -209,37 +201,41 @@ def _create_notebook_model(self):
     def test_delete_file(self):
         file_path = "/test_delete_exits_file.txt"
         message = "Lorem ipsum dolor sit amet..."
-        self.file_api.write_file(file_path, message, self.endpoint)
-
-        self.contents_manager.delete_file(file_path)
+        try:
+            self.file_api.write_file(file_path, message, self.endpoint)
+        finally:
+            self.contents_manager.delete_file(file_path)
 
         with self.assertRaises(IOError):
             self.file_api.stat(file_path, self.endpoint)
 
     def test_delete_non_exits_file(self):
         file_path = "/test_delete_non_exits_file.txt"
-
         with self.assertRaises(web.HTTPError):
             self.contents_manager.delete_file(file_path)
 
     def test_rename_file(self):
+        file_renamed = False
         file_path = "/test_rename_file.txt"
         message = "Lorem ipsum dolor sit amet..."
-        self.file_api.write_file(file_path, message, self.endpoint)
-
         file_dest = "/test_after_rename_file.txt"
 
-        self.contents_manager.rename_file(file_path, file_dest)
+        self.file_api.write_file(file_path, message, self.endpoint)
+        try:
+            self.contents_manager.rename_file(file_path, file_dest)
+            file_renamed = True
 
-        stat_info = self.file_api.stat(file_dest, self.endpoint)
-        self.assertIsInstance(stat_info, dict)
+            stat_info = self.file_api.stat(file_dest, self.endpoint)
+            self.assertIsInstance(stat_info, dict)
 
-        with self.assertRaises(IOError):
-            self.file_api.stat(file_path, self.endpoint)
+            with self.assertRaises(IOError):
+                self.file_api.stat(file_path, self.endpoint)
 
-        self.file_api.remove(file_dest, self.endpoint)
-        with self.assertRaises(IOError):
-            self.file_api.stat(file_dest, self.endpoint)
+            self.file_api.remove(file_dest, self.endpoint)
+            with self.assertRaises(IOError):
```

Comment:
I think stating here is unnecessary, remove should be moved to the finally block